static/js/components/light-capability.js

```diff
@@ -0,0 +1,290 @@
+      background-color: #89b6d6;
+    }
+
+    .webthing-light-capability-container.on {
+      background-color: white;
+      color: #5d9bc7;
+    }
+
+    .webthing-light-capability-info {
+      position: absolute;
+      top: 50%;
+      left: 50%;
+      transform: translate(-50%, -50%);
+    }
+
+    .webthing-light-capability-icon {
+      fill: white;
+    }
+
+    .webthing-light-capability-icon.bright {
+      stroke: #666;
+    }
+
+    .webthing-light-capability-label {
+      padding-top: 0.2rem;
+      font-weight: bold;
+    }
+  </style>
+  <div id="container" class="webthing-light-capability-container">
+    <div id="info" class="webthing-light-capability-info">
+      <svg id="icon" class="webthing-light-capability-icon"
+        xmlns="http://www.w3.org/2000/svg" height="64" width="64">
+        <g transform="translate(0,-988.36216)">
+          <path
+             d="m 41.6997,1041.6985 c 0,1.0723 -0.8727,1.9367 -1.9366,1.9367 l
+             -15.5179,0 c -1.0722,0 -1.9366,-0.8727 -1.9366,-1.9367 0,-1.0722
+             0.8727,-1.9366 1.9366,-1.9366 l 15.5179,0 c 1.0639,-0.01
+             1.9366,0.8644 1.9366,1.9366 z m -1.9449,2.9091 -15.5096,0 c
+             -1.28,0 -2.2608,1.2302 -1.8369,2.5683 0.2577,0.8063 1.0722,1.305
+             1.92,1.305 l 0.033,0 c 1.1221,0 2.1444,0.6317 2.6431,1.6374 l
+             0.017,0.041 c 0.6732,1.3465 2.053,2.2026 3.5657,2.2026 l 2.826,0 c
+             1.5127,0 2.8925,-0.8561 3.5657,-2.2026 l 0.017,-0.041 c
+             0.4987,-1.0057 1.5294,-1.6374 2.6432,-1.6374 l 0.033,0 c 0.8478,0
+             1.6623,-0.4987 1.92,-1.305 0.4322,-1.3381 -0.5569,-2.5683
+             -1.8369,-2.5683 z m 1.9449,-7.7631 c 0,1.0722 -0.8727,1.9366
+             -1.9366,1.9366 l -15.5179,0 c -1.0722,0 -1.9366,-0.8727
+             -1.9366,-1.9366 0,-1.0306 0.8062,-1.8701 1.8285,-1.9283
+             -1.2135,-10.9132 -12.5008,-13.3403 -12.5008,-26.19 0,-11.24577
+             9.118,-20.36377 20.3637,-20.36377 11.2457,0 20.3637,9.118
+             20.3637,20.36377 0,12.8497 -11.2873,15.2768 -12.4925,26.19
+             1.014,0.058 1.8285,0.8977 1.8285,1.9283 z" />
+        </g>
+      </svg>
+      <div id="label" class="webthing-light-capability-label">OFF</div>
+    </div>
+  </div>
+`;
+
+const OFF_FILL = '#ffffff';
+const ON_FILL = '#5d9bc7';
+
+class LightCapability extends BaseComponent {
+  constructor() {
+    super(template);
+
+    this._container = this.shadowRoot.querySelector('#container');
+    this._icon = this.shadowRoot.querySelector('#icon');
+    this._label = this.shadowRoot.querySelector('#label');
+
+    this._haveBrightness = false;
+    this._haveColor = false;
+    this._haveColorTemperature = false;
+    this._on = false;
+    this._brightness = 0;
+    this._color = OFF_FILL;
+    this._colorTemperature = 2700;
+
+    this._onClick = this.__onClick.bind(this);
+  }
+
+  connectedCallback() {
+    this._haveBrightness =
+      typeof this.dataset.haveBrightness !== 'undefined' ?
```

Comment:
Optional, but a thought I had reading this: most of these conditions could be simplified as `!!this.dataset.haveBrightness` since the main result of the ternary operator is coercion of undefined to boolean.

Reply:
This was actually invalid anyway. I needed to parse the string into a Boolean.